src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/Serialization/TagHelperDescriptorJsonConverter.cs

```diff
@@ -24,10 +26,35 @@ public override object ReadJson(JsonReader reader, Type objectType, object exist
                 return null;
             }
 
+            // Try reading the optional hashcode
+            var hashWasRead = reader.TryReadNextProperty(RazorSerializationConstants.HashCodePropertyName, out var hashObj);
+            var hash = hashObj as int?;
+            if (!DisableCachingForTesting &&
+                hashWasRead &&
+                hash.HasValue &&
```

Comment:
Isn't Hash guaranteed to have a value if it was read?

Reply:
Yep, just added it as an additional safeguard, removed.